### ISSUE TITLE
Str::excerpt does not always clip text at word spaces

### ISSUE THREAD

**Issue by frankrausch** [getkirby/kirby#2420]
**Describe the bug**  
`Str::excerpt()` sometimes returns partial words instead of always clipping to full words.
It seems to be related to the string containing characters outside of the ASCII range.

**To Reproduce**  

```php
<?php echo Str::excerpt('Kirby – ein großartiges CMS!', 20) ?>
/* returns "Kirby – ein g …" */
```

**Expected behavior**  
Should clip the text after full words only.
```php
<?php echo Str::excerpt('Kirby – ein großartiges CMS!', 20) ?>
/* should return "Kirby – ein …" */
```

**Potential Fix**
```php
// Str.php, line 364
return static::substr($string, 0, mb_strrpos(static::substr($string, 0, $chars), ' ')) . ' ' . $rep;
//                                ↑ replacing strrpos() with mb_strrpos() fixes this for me
```

**Kirby Version**  
3.3.3

**Additional context**  
PHP 7.4.1 on macOS 10.15.2

**Comment by afbora** [comment]
Your fix absolutely correct! `strrpos` method should be multibyte to support unicode chars.

**Comment by distantnative** [comment]
✅ Will be fixed in 3.3.4
